File: Code/Overloading.hs

data Square = Square { length :: Integer } deriving Show
data Rectangle = Rectangle { width :: Integer, height :: Integer } deriving Show

getArea :: Square -> Integer
getArea (Square l) = l*l

getArea :: Rectangle -> Integer
getArea (Rectangle w h) = w*h

main = do
    let s = Square 5
    let r = Rectangle 3 4
    putStrLn(show(getArea s))
    putStrLn(show(getArea r))

File: Code/Overloading.java
class Rectangle{
    private double width;
    private double height;

    public Rectangle(double width, double height){
        this.width = width;
        this.height = height;
    }

    public double getWidth(){
        return this.width;
    }

    public double getHeight(){
        return this.height;
    }

}

class Square{
    private double side;

    public Square(double side){
        this.side = side;
    }

    public double getSide(){
        return this.side;
    }

}

// Class for getting area
/**
 * Overloading
 */
public class Overloading {

    public static double getArea(Rectangle r){
        return r.getWidth() * r.getHeight();
    }

    public static double getArea(Square s){
        return s.getSide() * s.getSide();
    }

    public static void main(String[] args) {
        Rectangle r = new Rectangle(5, 10);
        Square s = new Square(5);

        System.out.println("Area of rectangle: " + getArea(r));
        System.out.println("Area of square: " + getArea(s));
    }

}